markdown_code_symlinks/__init__.py

```diff
@@ -106,20 +108,45 @@ def relative_docs(cls, url):
 
     @classmethod
     def add_mapping(cls, docs_rel, code_rel):
-        assert docs_rel not in cls.mapping['docs2code'], """\
+        def _sphinx_ge_3_0(cls, docs_rel, code_rel):
+            code_rel_no_ext = os.path.splitext(code_rel)[0]
+
+            assert docs_rel not in cls.mapping['docs2code'], """\
 Assertion error! Document already in mapping!
     New Value: {}
 Current Value: {}
 """.format(docs_rel, cls.mapping['docs2code'][docs_rel])
+            assert code_rel_no_ext not in cls.mapping['code2docs'], """\
+Assertion error! Document already in mapping!
+    New Value: {}
+Current Value: {}
+""".format(docs_rel, cls.mapping['code2docs'][code_rel])
```

Comment:
There are at least four of these prints. Probably it would be good to wrap this string formatting somewhere, maybe by defining the string beforehand and then formatting it when needed.

Reply:
Fixed